source/Stargate-Examples-Tests/SouthAmericanCurrenciesRESTfulControllerTest.class.st

```diff
@@ -74,6 +101,74 @@ SouthAmericanCurrenciesRESTfulControllerTest >> testGetArgentinePesoBanknotes [
 		do: [ :banknotes | self assertCollection: banknotes hasSameElements: #(5 10 20 50 100 200 500 1000) ]
 ]
 
+{ #category : #tests }
+SouthAmericanCurrenciesRESTfulControllerTest >> testGetArgentinePesoInEnglish [
+
+	| response request |
+
+	request := self
+		requestToGETResourceIdentifiedBy: 'ARS'
+		accepting: resourceController currencyVersion1dot0dot0MediaType.
+	request setAcceptLanguage: 'en'.
+
+	response := resourceController currencyBasedOn: request within: self newHttpRequestContext.
+
+	self
+		assert: response isSuccess;
+		assert: response status equals: 200;
+		assert: response contentType asMediaType
+			equals: resourceController currencyVersion1dot0dot0MediaType;
+		withTheOnlyOneIn: response contentLanguageTags
+			do: [ :tag | self assert: tag equals: 'en-US' asLanguageTag ];
+		assert: response varyHeaderNames includes: 'Accept-Language';
+		assert: response varyHeaderNames includes: 'Accept';
+		assert: response entityTag equals: '"5dd07a40a75ea23fa44e641a92a6dd1ec7999a36"' asETag.
+
+	self
+		withJsonFromContentsIn: response
+		do: [ :json | 
+			self
+				assert: json name equals: 'Argentine peso';
+				assert: json symbol equals: '$';
+				assert: json isoCode equals: 'ARS';
+				assert: json links isNil
+			]
+]
+
+{ #category : #tests }
+SouthAmericanCurrenciesRESTfulControllerTest >> testGetArgentinePesoInSpanish [
+
+	| response request |
+
+	request := self
+		requestToGETResourceIdentifiedBy: 'ARS'
+		accepting: resourceController currencyVersion1dot0dot0MediaType.
+	request setAcceptLanguage: 'es'.
+
+	response := resourceController currencyBasedOn: request within: self newHttpRequestContext.
+
+	self
+		assert: response isSuccess;
+		assert: response status equals: 200;
+		assert: response contentType asMediaType
+			equals: resourceController currencyVersion1dot0dot0MediaType;
+		withTheOnlyOneIn: response contentLanguageTags
+			do: [ :tag | self assert: tag equals: 'es-AR' asLanguageTag ];
+		assert: response varyHeaderNames includes: 'Accept-Language';
+		assert: response varyHeaderNames includes: 'Accept';
+		assert: response entityTag equals: '"fd65007599711248b85d86917bd24c5aa29e610b"' asETag.
+
+	self
+		withJsonFromContentsIn: response
+		do: [ :json | 
+			self
+				assert: json name equals: 'Peso';
+				assert: json symbol equals: '$';
+				assert: json isoCode equals: 'ARS';
+				assert: json links isNil
```

Comment:
If there is more than one place where we are checking for nil, this logic could be extracted and a comment added explaining the reason.